File: 백준/Silver/1920. 수 찾기/수 찾기.cc
#include <bits/stdc++.h>

using namespace std;

int number[10000001],m,question[100001];
long long int n;

int Bsearch(int target,int low,long long int high)
{
    if(low>high)
        return 0;
    int mid=(low+high)/2;
    if(number[mid]==target)
        return 1;
    else if(number[mid]>target)
        return Bsearch(target,low,mid-1);
    else
       return Bsearch(target,mid+1,high);
}

bool arr(int x,int y)
{
    return x<y;
}

int main()
{
    scanf("%d",&n);
    for(int i=0;i<n;i++){
        scanf("%d",&number[i]);
    }
    sort(number,number+n,arr);
    scanf("%d",&m);
    for(int i=0;i<m;i++){
        scanf("%d",&question[i]);
    }
    for(int i=0;i<m;i++){
        printf("%d\n",Bsearch(question[i],0,n-1));
    }
    return 0;
}

File: 백준/Silver/1920. 수 찾기/수 찾기.py
n = int(input())
nums = list(map(int, input().split()))
m = int(input())
targets = list(map(int, input().split()))

nums.sort()

for i in targets:
    l = 0
    r = n - 1
    while l <= r:
        mid = (l + r) // 2
        if nums[mid] == i:
            print(1)
            break
        elif nums[mid] > i:
            r = mid - 1
        else:
            l = mid + 1
    else:
        print(0)